### DISCUSSION TITLE
Evaluate the variance explanation of Bray-Curtis distance from one abundance matrix to another abundance matrix ?

### DISCUSSION THREAD

**Discussion by ChaoXianSen** [vegandevs/vegan#698] [Q&A]
```r
library(vegan)
data(varespec)
data(varechem)

# Bray-Curtis
veg.dist <- vegdist(varespec,method = "bray") 
env.dist <- vegdist(scale(varechem), method = "bray")

#  Evaluate the variance explanation of the Bray-Curtis distance from varechem abundance matrix to varespec abundance matrix
adonis_result<- adonis(veg.dist ~ as.matrix(env.dist), permutations = 999)

R2 <- adonis_result$aov.tab[1, "R2"]  
PValue <- adonis_result$aov.tab[1, "Pr(>F)"] 
```

Question:  Evaluate the variance explanation of the Bray-Curtis distance from varechem abundance matrix to varespec abundance matrix,  Can I use this holistic assessment method?   Looking forward to your reply.

**Comment by jarioksa** [comment]
There is a problem in your model definition: you shall **not** use dissimilarities on the right-hand-side of `adonis2` (or outdated and soon defunct `adonis`) model, but you must use rectangular data matrix:
```r
adonis.result <- adonis2(veg.dist ~ ., data=varechem, permutations=999)
```
It makes no sense to use distances on the right-hand-side. You must use rectangular data

The reported _R_<sup>2</sup> has been interpreted similarly as variance explained. Strictly speaking it is not variance, but it is squared Bray-Curtis distance that is decomposed instead of variance.

**Comment by ChaoXianSen** [comment]
![image](https://github.com/user-attachments/assets/43062ce8-f55e-4866-8718-4d637692cf07)
![image](https://github.com/user-attachments/assets/4d91019e-fedf-46a0-9d07-eb6f3f074952)
They all seem to work, but the results are inconsistent.

**Reply by gavinsimpson** [reply]
What do you mean by "inconsistent"? If you mean the results vary from run to run of the code, that is to be expected as the tests are based on producing the sampling distribution of the test statistic under the Null hypothesis ($\text{H}_0$) using **permutation** of the data. This is a (pseudo)random process. If you want to get the same result from run to run you *must* set a seed for the RNG and run that code to set the seed before each run:

```{r}
set.seed(1) # essentially any number in here
adonis2(....) # run your adonis2 model
```

The run-to-run variability (without setting a seed) will tend to 0 as you increase the number of permutations used for the sampling distribution of the test statistic under $\text{H}_0$.

**Reply by gavinsimpson** [reply]
Also, @jarioksa said that it makes no sense to use ` ~ as.matrix(env.dist)`, so I'm not sure why you posted that first image? We can't read you mind, so it would be helpful to explain why you are posting something. Also, FYI: you can copy and paste text from the R console into your posts here on GH. Just mark them as code by putting ````` ```{r} ````` on the line above the code and ```` ``` ```` on the line below the code/output. block

**Reply by ChaoXianSen** [reply]
@jarioksa said that it makes no sense to use  ~ as.matrix(env.dist)， But  “adonis2(veg.dist ~ as.matrix(env.dist), permutations = 999)” this way you can calculate the result. So I can understand that although the results can be calculated in this way, the results of the calculation are not of any explanatory significance，Is this understanding OK?

**Reply by gavinsimpson** [reply]
Where does Jari say *that*?!

Putting aside the fact that Bray Curtis is not a good choice for *environmental* data measured in different units --- your scaling via `scale()` to get round this issue is causing a warning that says:

```
  results may be meaningless because data have negative entries
                 in method “bray”
```

--- when I run your code I get a 0 residual component thus $R^2$ is 100% (with `adonis2()`. So, No, I don't think you understood what Jari said.

If you update your vegan to the latest version you should see this output by default now:

```{r}
Permutation test for adonis under reduced model
Permutation: free
Number of permutations: 999

adonis2(formula = veg.dist ~ ., data = varechem, permutations = 999)
         Df SumOfSqs      R2      F Pr(>F)
Model    14   3.1740 0.69843 1.4888  0.072 .
Residual  9   1.3705 0.30157
Total    23   4.5444 1.00000
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
```

where the $R^2$ is the proportion of the *squared* Bray Curtis distance of the **response** that is explained by the 14 env variables in `varechem`.

You didn't respond to my other question asking what you mean by "inconsistent". I'll ask that again. But if you mean't why do these two approaches (your two images) give inconsistent results, it's because you misunderstood what Jari said and did exactly the thing he told you not to do, which is use a distance matrix as explanatory term in `adonis2()`. The way to do this is to follow Jari's reply and use:

```{r}
adonis.result <- adonis2(veg.dist ~ ., data=varechem, permutations=999)
```

The way you are doing it is wrong.
